src/wp-includes/class-wp-block.php

```diff
@@ -514,6 +524,8 @@ public function render( $options = array() ) {
 						/** This filter is documented in wp-includes/blocks.php */
 						$inner_block->context = apply_filters( 'render_block_context', $inner_block->context, $inner_block->parsed_block, $parent_block );
 
+						$inner_block->update_available_context( $inner_block->parsed_block, $inner_block->context );
```

Comment:
Is this called here because the values are filtered above? If so, maybe we should first check whether the filtered values are different from the original before calling this method?

Reply:
Yes, that would be a good enhancement.

Reply:
Nice catch. Check added in https://github.com/WordPress/wordpress-develop/pull/7522/commits/5cbb114252c844881ed02da768a7b48108768e57